src/main/java/uk/gov/justice/probation/courtcasematcher/messaging/HearingExtractor.java

```diff
@@ -63,11 +71,26 @@ List<Hearing> extractHearings(String payload, String messageId) {
     }
 
     private List<Hearing> parseCPMessage(SnsMessageContainer snsMessageContainer) throws JsonProcessingException {
-        final var cpHearingEvent = commonPlatformParser.parseMessage(snsMessageContainer.getMessage(), CPHearingEvent.class);
+        String message = snsMessageContainer.getMessage();
+
+        if (snsMessageContainer.getMessageAttributes().getExtendedPayloadSize() != null) {
+            message = getPayloadFromS3(snsMessageContainer);
+            log.info("Successfully downloaded large message from s3 bucket with extended payload size: {}", snsMessageContainer.getMessageAttributes().getExtendedPayloadSize());
+        }
+        final var cpHearingEvent = commonPlatformParser.parseMessage(message, CPHearingEvent.class);
         final var hearing = cpHearingEvent.asDomain();
         return passHearingIdToCourtCaseService ? setHearingAttributes(hearing, cpHearingEvent, snsMessageContainer) : hearing;
     }
 
+    private String getPayloadFromS3(SnsMessageContainer snsMessageContainer) throws JsonProcessingException {
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
Have you considered using the ObjectMapper bean?
https://github.com/ministryofjustice/court-case-matcher/blob/65f7cd29a9439855ec952f58b7aaa1e84dc4206b/src/main/java/uk/gov/justice/probation/courtcasematcher/application/MessagingConfig.java#L28